metadata-ingestion/src/datahub/ingestion/api/source.py

```diff
@@ -219,36 +236,82 @@ def failures(self) -> LossyList[StructuredLogEntry]:
     def infos(self) -> LossyList[StructuredLogEntry]:
         return self._structured_logs.infos
 
+    def _store_workunit_data(self, wu: MetadataWorkUnit) -> None:
```

Comment:
We should not be mixing all of these different capabilities into the same class. Ideally, in this case, we should instead have a SubtypesReport that the main SourceReport includes as a field. That type should also have a compute_stats method. That way we would also be able to unit test the SubtypeReport more easily. 

Open to changes on naming e.g. SubtypeCountReport or something would also be fine